core/src/test/scala/unit/kafka/server/ClientQuotaManagerTest.scala

```diff
@@ -49,6 +53,27 @@ class ClientQuotaManagerTest {
                           new Quota(2000, true), clientMetrics.quota("p1"))
       Assert.assertEquals("Should return the overridden value (4000)",
                           new Quota(4000, true), clientMetrics.quota("p2"))
+
+      // p1 should be throttled using the default quota
+      var throttleTimeMs = clientMetrics.recordAndMaybeThrottle("p1", 2500 * config.numQuotaSamples, this.callback)
+      Assert.assertTrue(s"throttleTimeMs should be > 0. was $throttleTimeMs", throttleTimeMs > 0)
+
+      // Case 2: Change quota again. The quota should be updated within KafkaMetrics as well since the sensor was created.
+      // p1 should not longer be throttled after the quota change
+      clientMetrics.updateQuota("p1", new Quota(3000, true));
+      Assert.assertEquals("Should return the newly overridden value (3000)",
+        new Quota(3000, true), clientMetrics.quota("p1"))
+
+      throttleTimeMs = clientMetrics.recordAndMaybeThrottle("p1", 0, this.callback)
```

Comment:
The idea was to change the quota and test that the throttle time is 0 for the new quota. 
Recording a 0 doesn't change the underlying value. It is basically trying to verify that new quota is applied and doesn't throttle on the previously recorded value.